week1/645_set_mismatch.py

```diff
@@ -0,0 +1,11 @@
+# https://leetcode.com/problems/set-mismatch/
+def set_mismatch(nums: list[int]) -> list[int]:
+    freq_dict = {i + 1: 0 for i in range(len(nums))}
+    for n in nums:
+        freq_dict[n] += 1
+
+    freq_list = list(freq_dict.values())
+    if 0 not in freq_list:
+        return []
+    else:
+        return [freq_list.index(2) + 1, freq_list.index(0) + 1]
```

Comment:
This loop (caused by `.index` can be avoidable. Can you think of a way that you can solve this mathematically?

Reply:
Could you maybe give some hints? I cannot think of any mathematical method to find a key from the dictionary with the value without iterating through all the elements (so worst case O(n)?).

Reply:
> This loop (caused by `.index` can be avoidable. Can you think of a way that you can solve this mathematically?

Could you maybe give some hints? I cannot think of any mathematical method to find a key from the dictionary with the value without iterating through all the elements (so worst case O(n)?).

Reply:
I could try list comprehension and another for loops but it is also just a loop. Did you mean that I should try to solve from top to bottom differently?

Reply:
> I could try list comprehension or another for loop but it is also just a loop. Did you mean that I should try to solve from top to bottom differently?

Sorry, you're right. You can avoid one `.index()` but not both. The time complexity in terms of bigO notation should be same even with the improvements, but you can still reduce the number of operations.